src/app/models/form.server.model.ts

```diff
@@ -559,6 +559,24 @@ const compileFormModel = (db: Mongoose): IFormModel => {
     return this.save()
   }
 
+  FormDocumentSchema.methods.duplicateFormFieldById = function (
+    this: IFormDocument,
+    fieldId: string,
+  ) {
+    const fieldToDuplicate = getFormFieldById(this.form_fields, fieldId)
+    if (!fieldToDuplicate) return Promise.resolve(null)
+    const duplicatedField = JSON.parse(JSON.stringify(fieldToDuplicate))
+    // Remove unique ids before saving
+    delete duplicatedField.globalId
+    delete duplicatedField._id
```

Comment:
you're doing this to avoid mutating the original field, right? you can consider using lodash's [omit](https://lodash.com/docs/4.17.15#omit) to do this instead